### PR TITLE
Add global key support for Project move_data_rows_to_task_queue

### PR DESCRIPTION
**Pull request by vbrodsky**
Story: https://labelbox.atlassian.net/browse/SDK-414
API PR: https://github.com/Labelbox/intelligence/pull/18036

- We are implementing the `Identifiable` hierarchy to ensure a consistent approach in handling data row IDs and global keys.
- To support both legacy string inputs and the new `Identifiable` objects, we are utilizing the @overload decorator. 
- Our plan is to apply this same methodology to other methods that require the management of both data row IDs and global keys.

### CODE REVIEW

File: labelbox/schema/project.py
Comment (vbrodsky) [inline labelbox/schema/project.py:642]: we have gotten rid of this monkeypatch sometime ago

File: labelbox/schema/identifiable.py
Comment (kopreschko) [inline labelbox/schema/identifiable.py:4]: TypeAlias seems to be available since 3.10

Comment (vbrodsky) [reply labelbox/schema/identifiable.py:4]: yes, but I am using `typing_extensions` that back-ported it. However `TypeAlias` is not strictly necessary, it is just for improved typing. Should I remove it?

File: labelbox/schema/identifiable.py
Comment (vbrodsky) [reply labelbox/schema/identifiable.py:18]: I am setting it now in derived classes, but why should it be in the constructor? Actually I  think it should be a const

File: labelbox/schema/identifiable.py
Comment (vbrodsky) [reply labelbox/schema/identifiable.py:35]: I wanted to keep it as a method in case we want to amend the `keys` getter in the future and it will then keep the semantics for `__eq__`

File: labelbox/schema/identifiable.py
Comment (vbrodsky) [reply labelbox/schema/identifiable.py:62]: ok that is cool

File: labelbox/schema/identifiable.py
Comment (kopreschko) [reply labelbox/schema/identifiable.py]: I can think of at least two reasons:
- the inclusion of `id_type` in the constructor of `Identifiable` explicitly communicates the necessity of specifying the type when constructing an Identifiable object
- `Identifiable` should refrain from making assumptions about the default type. Assuming a default type without explicit assignment by derived classes could lead to potential issues, especially when new derived classes are introduced in the future.

**Review by kopreschko — APPROVED** [review]
lgtm

File: labelbox/schema/identifiable.py
Comment (kopreschko) [inline labelbox/schema/identifiable.py:65]: missing a newline at the end of the file

Comment (whistler) [reply labelbox/schema/identifiable.py:65]: From a user point of view, it does not seem clear that UniqueIds is DataRowIds, since global keys are also unique and they are also ids. What do you think of creating a sub-class / alias for `UniqueIds` called `DataRowIds`

Comment (vbrodsky) [reply labelbox/schema/identifiable.py:65]: Hi @whistler I am trying to avoid 'over-classing' but I see your point. Would something like this help
`DataRowIds = UniqueIds` i.e. kind of class alias
PS We will also document this in a number of places

**Comment by adrian-chang** [comment]
LGTM, i'll wait on the engineers to give the final approval

File: labelbox/schema/identifiable.py
Comment (sqlboy) [inline labelbox/schema/identifiable.py:7]: in ADV these are ID and GKEY, these are used by the census / bucket scanning APIS.

File: labelbox/schema/identifiable.py
Comment (sqlboy) [inline labelbox/schema/identifiable.py:11]: Confused why this is a wrapper around an array, from initial designs this is basically a wrapper around a single id.

File: labelbox/schema/identifiable.py
Comment (sqlboy) [reply labelbox/schema/identifiable.py:65]: Confused why this code even exists really.

Comment (whistler) [reply labelbox/schema/identifiable.py:65]: Yes @vbrodsky, that seems good to me.

File: labelbox/schema/identifiable.py
Comment (sqlboy) [reply labelbox/schema/identifiable.py:11]: Maybe its a matter of renaming it to Identifiables

Comment (sqlboy) [reply labelbox/schema/identifiable.py:11]: The idea with Identifiable ABC is that its an interface that lets code know the object is an entity with a unique Id and it provides some function for code to fetch the id structure.  This is more like a utility class, but since its a list you should implement __iter__ and other functions so it can act like a collection.

File: labelbox/schema/identifiable.py
Comment (vbrodsky) [reply labelbox/schema/identifiable.py:7]: ok I can change it, but in a follow up PR because this will require the api update too

File: labelbox/schema/identifiable.py
Comment (sqlboy) [inline labelbox/schema/identifiable.py:31]: Here there is no reason to duplicate the argument type being passed in.  (strings_). The function params already says its strings.  Many libs use like of() or from(), like:

```
keys  = GlobalKeys.of(["some key"])
```

For example in spanner library, its ```Key.of("value")```

File: labelbox/schema/identifiable.py
Comment (vbrodsky) [reply labelbox/schema/identifiable.py:5]: Implemented as an iter, that is a good idea thank you @sqlboy

File: labelbox/schema/identifiables.py
Comment (vbrodsky) [reply labelbox/schema/identifiables.py:1]: of course

**Review by sqlboy — APPROVED** [review]

**Pull request merged**
